### ISSUE TITLE
false positive of `Detected running in editor,`

### ISSUE THREAD

**Issue by ryoppippi** [antfu/eslint-config#590]
### Describe the bug

When I run eslint command, the message appears.
```
[@antfu/eslint-config] Detected running in editor, some rules are disabled.
```

version v2.27.1 [Latest](https://github.com/antfu/eslint-config/releases/latest)

For reproduction, 
1. download the repository below into local environment
2. `pnpm i` 
3. `pnpm format`

and you can see the message above

### Reproduction

https://stackblitz.com/edit/vitejs-vite-6cpmmz?file=package.json

### System Info

```Shell
System:
    OS: macOS 14.5
    CPU: (11) arm64 Apple M3 Pro
    Memory: 424.66 MB / 36.00 GB
    Shell: 3.7.1 - /opt/homebrew/bin/fish
  Binaries:
    Node: 20.15.1 - ~/.local/share/devbox/global/default/.devbox/nix/profile/default/bin/node
    npm: 10.7.0 - ~/.local/share/devbox/global/default/.devbox/nix/profile/default/bin/npm
    bun: 1.1.24 - ~/.local/share/aquaproj-aqua/bin/bun
  Browsers:
    Chrome: 127.0.6533.120
    Safari: 17.5
```

### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/antfu/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/antfu/contribute).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

### Contributions

- [X] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests (actually just go ahead and do it, thanks!)

**Comment by ryoppippi** [comment]
I found that in my shell, `process.env.NVIM` is always has value.
This might causes a problem

**Comment by ryoppippi** [comment]
This only happens in my env, so close this
